skbio/stats/composition.py

```diff
@@ -2058,3 +2061,407 @@ def dirmult_ttest(
+        else:
+            qval = single_covar_data["pvalue"].values
+
+        single_covar_data["qvalue"] = qval[0]
+
+    # Creating an empty dict to store sum of values (Using a DataFrame threw errors)
+    # uses a separate array for each covariate and each feature
+    # array index: 0: pvalue, 1: CI(2.5), 2: CI(97.5), 3: qvalue, 4: log2fc
+    res_dict = {}
+
+    for b in data.columns:
+        res_dict[b] = {}
+        for covar in list_of_vars:
+            res_dict[b][covar] = [0, 0, 0, 0, 0]
+
+    for single_covar_data in res:
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][0] = (
+            single_covar_data["pvalue"]
+        )
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][1] = (
+            single_covar_data["CI(2.5)"]
+        )
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][2] = (
+            single_covar_data["CI(97.5)"]
+        )
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][3] = (
+            single_covar_data["qvalue"]
+        )
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][4] = (
+            single_covar_data["Log2(FC)"]
+        )
+
+    for i in range(1, draws):
+        posterior = [
+            rng.dirichlet(data.values[i] + pseudocount) for i in range(data.shape[0])
+        ]
+        dir_table = pd.DataFrame(clr(posterior), index=data.index, columns=data.columns)
+
+        _call_ires = _lme_call(
+            formula=formula,
+            table=dir_table,
+            metadata=metadata,
+            groups=groups,
+            reml=reml,
+            method=method,
+            fit_kwargs=fit_kwargs,
+            **kwargs,
+        )
+
+        ires = _call_ires[0]
+
+        if ires is None:
+            continue
+
+        for single_covar_data in ires:
+            # multiple comparison
+            if p_adjust is not None:
+                qval = _calc_p_adjust(p_adjust, single_covar_data["pvalue"])
+            else:
+                qval = single_covar_data["pvalue"].values
+
+            single_covar_data["qvalue"] = qval[0]
+
+        # TODO: online average to avoid holding all of the results in memory
+        for single_covar_data in ires:
+            curr_feature_id = single_covar_data["FeatureID"]
+            curr_covariate = single_covar_data["Covariate"]
+
+            res_dict[curr_feature_id][curr_covariate][0] += single_covar_data["pvalue"]
+            res_dict[curr_feature_id][curr_covariate][1] += single_covar_data["CI(2.5)"]
+            res_dict[curr_feature_id][curr_covariate][2] += single_covar_data[
+                "CI(97.5)"
+            ]
+            res_dict[curr_feature_id][curr_covariate][3] += single_covar_data["qvalue"]
+            res_dict[curr_feature_id][curr_covariate][4] += single_covar_data[
+                "Log2(FC)"
+            ]
+
+    # TODO: implement online averages to save memory
```

Comment:
Let's do this now. We don't want to save all of the bootstraps in memory.

See these lines on how to implement this
https://github.com/scikit-bio/scikit-bio/blob/main/skbio/stats/composition.py#L2019-L2024